provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/KotlinInitScriptIntegrationTest.kt

```diff
@@ -0,0 +1,30 @@
+package org.gradle.kotlin.dsl.integration
+
+import org.gradle.kotlin.dsl.fixtures.AbstractIntegrationTest
+import org.gradle.kotlin.dsl.fixtures.DeepThought
+import org.junit.Test
+
+
+class KotlinInitScriptIntegrationTest : AbstractIntegrationTest() {
+
+    @Test
+    fun `initscript classpath`() {
+
+        withClassJar("fixture.jar", DeepThought::class.java)
+
+        val initScript =
+            withFile("init.gradle.kts", """
+
+                initscript {
+                    dependencies { classpath(files("fixture.jar")) }
+                }
+
+                val computer = ${DeepThought::class.qualifiedName}()
+                val answer = computer.compute()
+                println("*" + answer + "*")
+            """)
+
+        assert(
+            build("-I", initScript.canonicalPath).output.contains("*42*"))
+    }
```

Comment:
Can you add a test that covers making sure that the `baseDir: File` is set in accordance to the behaviour of groovy based init scripts?